### ISSUE TITLE
Updates to advanced_smart_access and related

### ISSUE THREAD

**Issue by jere-software** [jeremiahbreeden/bullfrog#9]
https://github.com/jeremiahbreeden/bullfrog/blob/f4ddd4a1438edc2acdc0b54d156bc671896dc201/src/bullfrog-access_types-advanced_smart_access-make.ads#L32
Update description to specify package must be at the same accessibility level as parent and that only one instance should be created (or runtime errors)

https://github.com/jeremiahbreeden/bullfrog/blob/f4ddd4a1438edc2acdc0b54d156bc671896dc201/src/bullfrog-access_types-advanced_smart_access-make.adb#L275
Update comment not to reference anything about volatile
Make sure the access object isn't already set before assigning and raise an exception if it does

https://github.com/jeremiahbreeden/bullfrog/blob/f4ddd4a1438edc2acdc0b54d156bc671896dc201/src/bullfrog-access_types-advanced_smart_access.ads#L447
remove volatile aspect

https://github.com/jeremiahbreeden/bullfrog/blob/f4ddd4a1438edc2acdc0b54d156bc671896dc201/src/bullfrog-access_types-advanced_smart_access.ads#L302
https://github.com/jeremiahbreeden/bullfrog/blob/f4ddd4a1438edc2acdc0b54d156bc671896dc201/src/bullfrog-access_types-advanced_smart_access.ads#L310
update comment to forbid making objects of reference type (repeat for Smart_Access package)

https://github.com/jeremiahbreeden/bullfrog/blob/f4ddd4a1438edc2acdc0b54d156bc671896dc201/src/bullfrog-access_types-advanced_smart_access_traits.ads#L36
Move access type and atomic_increment choice to traits package.
